rawfile/analytics/ga4/__init__.py

```diff
@@ -1,6 +1,81 @@
 from .client import GA4Client
 from .event import OpenEBSEventBuilder
 from .usage import Usage
+from .ping import Ping
+from consts import CONFIG
+from utils.logs import logger
+from utils.units import parse_time_delta
+
+import threading
+import queue
+import time
+
+task_queue = queue.Queue()
+stop_event = threading.Event()
+
+
+def get_usage():
+    event_usage = Usage(
+        api_secret=CONFIG["ga_key"],
+        measurement_id=CONFIG["ga_id"],
+        nodeid=CONFIG["nodeid"],
+    )
+    return event_usage
+
+
+def worker():
+    # reload every 4h
+    cache_duration = 60 * 60 * 4
+    last_updated = 0
+    event_usage = get_usage()
+    while not stop_event.is_set():
+        task = task_queue.get()
+        if task is None:
+            break
+
+        current_time = time.time()
+        if current_time - last_updated > cache_duration:
+            event_usage.refresh()
+            last_updated = current_time
+            logger.debug("Updated cache")
+
+        try:
+            task(event_usage)
+        except Exception as e:
+            # todo retry?
+            logger.error("Event Worker Error", exception=e)
+            pass
+        task_queue.task_done()
+    logger.info("Event Worker Exited")
+
+
+def shutdown_event_worker():
+    stop_event.set()
+
+
+def send_event(event_fn):
```

Comment:
send_event = lambda event_fn : task_queue.put(event_fn)

Reply:
the linter complains about these lambda, asking me to add them as def (same thing on the enable)

Reply:
Ok, no problem

I have suggested them since they where some simple functions or aliases

Reply:
Just for completeness, adding here:
```
rawfile/analytics/ga4/__init__.py:52:1: E731 Do not assign a `lambda` expression, use a `def`
   |
52 | shutdown_event_worker = lambda : stop_event.set()
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E731
   |
   = help: Rewrite `shutdown_event_worker` as a `def`
```